lib/liquid_voting/delegations.ex

```diff
@@ -115,6 +115,18 @@ defmodule LiquidVoting.Delegations do
     |> Repo.insert()
   end
 
+  @doc """
+  Creates a delegation.
+
+  ## Examples
+
+      iex> create_delegation(%{field: value})
+      %Delegation{}
+
+      iex> create_delegation(%{field: bad_value})
+      Ecto.*Error
```

Comment:
This is great, although the function doesn't currently do what we expect as it's using `insert` without the bang (currently returns `{:error, changeset}`): 

https://github.com/liquidvotingio/api/blob/7da73d6e51bbfd7e565049efc390001edae2b07c/lib/liquid_voting/delegations.ex#L129-L135 

I've also noticed some inconstancies in the return value in our docs: sometimes it's `Ecto.*Error`, elsewhere in the codebase it's `** (Ecto.NoResultsError)`. We should be consistent and precise and use the output when running these in an `IEx` shell.

Reply:
You mean we use different forms for the same function? Because some functions may give different error(s) to `** (Ecto.NoResultsError)`, which is why they specify `Ecto.*Error`. Or you mean sometimes the wrong error type (of the 2 mentioned) is specified for a function?

Reply:
Ah gotcha, this looks good in that case—let's use the precise error when we know what could be returns (`get` for example) and the more general `Ecto.*error` when we can't be specific. I just wanted to make sure we were being precise with our errors.

Reply:
regarding your first point:


api/lib/liquid_voting/delegations.ex

Lines 129 to 135 in 7da73d6
   """ 
   def create_delegation!(attrs \\ %{}) do 
     %Delegation{} 
     |> Delegation.changeset(attrs) 
     |> Repo.insert() 
   end 
  """
change the error in @do, or change to `insert!`?

Reply:
We should have `insert!` there (there's already a clause of `create_delegation`, no bang)

Reply:
Also you seem to be pinging quite a few people using `@`...

Reply:
I suppose it's a marketing strategy 😂

Reply:
I know... still working on my GH communication skills. Gimme time.

Reply:
Made the 2 edits @niklaslong  suggested.